flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkInputFormat.java

```diff
@@ -76,9 +78,12 @@ public BaseStatistics getStatistics(BaseStatistics cachedStatistics) {
   public FlinkInputSplit[] createInputSplits(int minNumSplits) throws IOException {
     // Called in Job manager, so it is OK to load table from catalog.
     tableLoader.open();
+    final ExecutorService workerPool = ThreadPools.newWorkerPool("iceberg-plan-worker-pool", context.planParallelism());
     try (TableLoader loader = tableLoader) {
       Table table = loader.loadTable();
-      return FlinkSplitPlanner.planInputSplits(table, context);
+      return FlinkSplitPlanner.planInputSplits(table, context, workerPool);
+    } finally {
+      workerPool.shutdown();
```

Comment:
This function is called in client and job manager. So there is no context here. Given it's a adhoc pool and will be shut down after planning, i think it's ok to naming it in this way.